### ISSUE TITLE
yazi triggers search up copy mode in tmux on open

### ISSUE THREAD

**Issue by ynejmidev** [sxyazi/yazi#1621]
### What system are you running Yazi on?

Linux Wayland

### What terminal are you running Yazi in?

foot version: 1.18.1 +pgo +ime +graphemes -assertions

### `yazi --debug` output

```Shell
Yazi
    Version: 0.3.3 (Nixpkgs 2024-09-04)
    Debug  : false
    OS     : linux-x86_64 (unix)

Ya
    Version: 0.3.3 (Nixpkgs 2024-09-04)

Emulator
    Emulator.via_env: ("foot", "tmux")
    Emulator.via_csi: Ok(Foot)
    Emulator.detect : Foot

Adapter
    Adapter.matches: Sixel

Desktop
    XDG_SESSION_TYPE           : Some("wayland")
    WAYLAND_DISPLAY            : Some("wayland-1")
    DISPLAY                    : Some(":0")
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: Some("9a09eac79b85c846e3a865a9078a3f8ff65a9259_1725836669_984023958")
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

Variables
    SHELL              : Some("/run/current-system/sw/bin/fish")
    EDITOR             : Some("nvim")
    VISUAL             : None
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None

Text Opener
    default: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block  : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

Multiplexers
    TMUX               : true
    tmux version       : 3.4
    ZELLIJ_SESSION_NAME: None
    Zellij version     : No such file or directory (os error 2)

Dependencies
    file             : 5.45
    ueberzugpp       : 2.9.6
    ffmpegthumbnailer: 2.2.2
    magick           : 7.1.1-38
    fzf              : 0.55.0
    fd               : 10.2.0
    rg               : 14.1.0
    chafa            : 1.14.2
    zoxide           : 0.9.4
    7z               : No such file or directory (os error 2)
    7zz              : 24.08
    jq               : 1.7.1
```

### Did you try the latest nightly build to see if the problem got fixed?

No, and I'll explain why below

### Describe the bug

https://github.com/user-attachments/assets/cee09aae-892b-4854-89ae-68913753ac59

It looks like some character sequence outputted by yazi is being evaluated by tmux as keybinds and triggering copy mode and then '?' to search up.
I had allow-passthrough set to 'on' initially, but whether with or without it this behaviour persists..

I'm pretty sure it was fine a couple days ago.. I don't use this software often so I don't remember exactly when it started behaving this way or on which update.

As for the nightly build I don't see a package for it in nixpkgs so I tried `nix run`ing the flake, and the issue persists. I don't know if yazi nightly is the same as a build from master, if there's more tests I could do I'd be happy to help.

### Minimal reproducer

- add yazi module in home manager
- run tmux
- run yazi

**Comment by sxyazi** [comment]
I can’t reproduce it in my tmux. Have you tried reproducing it with the default tmux config?

```
❯ tmux -V
tmux 3.4
```

**Comment by ynejmidev** [comment]
oh yeah, I have a keybind in tmux for copy mode, `alt-[`, I removed it and the problem went away, is that expected behaviour? are alt keybinds not recommended in tmux?

**Comment by sxyazi** [comment]
I'm not quite sure, I haven't really used tmux myself and just installed it to test Yazi. Just searched through the tmux issue list but couldn't find anything related to the `Alt-l` key.
  
Anyway, I'm glad to see the issue has been resolved, let's close the issue then.

**Comment by ynejmidev** [comment]
not alt-l but alt left square bracket.
Godspeed.

**Comment by Todd-Fulton** [comment]
Same issue here, yazi is sending `Alt-[` to the terminal and tmux is receiving them. Yazi is also sending `?12;0$y1$r1 q0u62;c` to the terminal after starting.

If I have a window that tmux moves to open previously, it sends a few codes that look like key presses the terminal would receive. I have the `tilit` tmux plugin installed, which is what set that keybind to switch windows in tmux.

I just noticed this issue today.

**Comment by sxyazi** [comment]
> Yazi is also sending `?12;0$y1$r1 q0u62;c` to the terminal after starting.

No, it's the terminal replying to Yazi with `DECSET` and `DECRQM` responses, i.e. terminal sends to tmux, when you run `echo -e '\033[?12$p\033P$q q\033\\'; cat` directly in the terminal, you'll see this. It looks like tmux mistakenly identified it as a key event and forwarded it to Yazi.

> yazi is sending `Alt-[` to the terminal and tmux is receiving them

As a CLI tool, Yazi cannot send keypresses to the terminal or tmux. Instead, Yazi only receives them from tmux/terminal. In this case, tmux somehow passed the wrong key event to Yazi, causing Yazi to open the search box. 

Tmux can interfere with the communication between Yazi and the actual terminal in various ways, which is likely related to https://github.com/sxyazi/yazi/issues/1557, please read through it. In the PR linked in that issue, Yazi allows `DECSET` and `DECRQM` requests to passthrough tmux to prevent any interference. This is the only recent change related to it, but the strange thing is that this change was already included in version 0.3.2, I'm not sure why 0.3.2 worked for you

**Comment by Todd-Fulton** [comment]
Thanks, that was a bit of a rabbit hole. I think I have a better understanding of what's going. Yazi is requesting features from the terminal via printing codes, and tmux is interfering in some way resulting in keys "being pressed" and bindings (if any) being triggered, in my case `Alt-[` which changes to previous window, and then the remaining code is "being typed" (interpreted as more key presses by tmux) into the terminal (or which ever application is open in the previous window).
